### PR TITLE
load quic early to prevent interaction with tests

### LINKED ISSUE #75484: System.Net.Security crash on arm64-Mono
Frequency in [last 30 days](https://runfo.azurewebsites.net/search/tests/?q=started%3A%7E30+definition%3Aruntime+name%3A%22system.net.security.tests+work+item%22) in Runfo as of 9/12:

Day | Run | Details
-- | -- | --
9/11 | Rolling run [11861](https://runfo.azurewebsites.net/view/build/?number=11861) | net7.0-Linux-Release-**arm64-Mono**_release-(Ubuntu.1804.ArmArch.Open)Ubuntu.1804.ArmArch.Open
9/10 | PR [11765](https://runfo.azurewebsites.net/view/build/?number=11765) | net7.0-Linux-Debug-**arm64-Mono**_release-(Ubuntu.1804.ArmArch.Open)Ubuntu.1804.ArmArch.Open
9/9 | PR [11335](https://runfo.azurewebsites.net/view/build/?number=11335) | net7.0-Linux-Debug-**arm64-Mono**_release-(Ubuntu.1804.ArmArch.Open)Ubuntu.1804.ArmArch.Open
9/9 | PR [10792](https://runfo.azurewebsites.net/view/build/?number=10792) | net7.0-Linux-Debug-**arm64-Mono**_release-(Ubuntu.1804.ArmArch.Open)Ubuntu.1804.ArmArch.Open
9/9 | PR [10264](https://runfo.azurewebsites.net/view/build/?number=10264) | net7.0-Linux-Debug-**arm64-Mono**_release-(Ubuntu.1804.ArmArch.Open)Ubuntu.1804.ArmArch.Open
9/8 | Rolling run [9270](https://runfo.azurewebsites.net/view/build/?number=9270) | net7.0-Linux-Release-**arm64-Mono**_release-(Debian.11.Arm64.Open)Ubuntu.1804.Armarch.Open
9/8 | PR [8767](https://runfo.azurewebsites.net/view/build/?number=8767) | net7.0-Linux-Debug-**arm64-Mono**_release-(Debian.11.Arm64.Open)Ubuntu.1804.Armarch.Open &<br/>net7.0-Linux-Debug-**arm64-Mono**_release-(Ubuntu.1804.ArmArch.Open)Ubuntu.1804.ArmArch.Open &<br/>net7.0-Linux-Debug-**arm64-Mono**_release-(Ubuntu.1804.ArmArch.Open)Ubuntu.1804.ArmArch.Open
9/7 | PR [8334](https://runfo.azurewebsites.net/view/build/?number=8334) | net7.0-Linux-Debug-**arm64-Mono**_release-(Debian.11.Arm64.Open)Ubuntu.1804.Armarch.Open
9/7 | PR [8314](https://runfo.azurewebsites.net/view/build/?number=8314) | net7.0-Linux-Debug-**arm64-Mono**_release-(Ubuntu.1804.ArmArch.Open)Ubuntu.1804.ArmArch.Open
9/7 | PR [8197](https://runfo.azurewebsites.net/view/build/?number=8197) | net7.0-Linux-Debug-**arm64-Mono**_release-(Ubuntu.1804.ArmArch.Open)Ubuntu.1804.ArmArch.Open &<br/>net7.0-Linux-Debug-**arm64-Mono**_release-(Ubuntu.1804.ArmArch.Open)Ubuntu.1804.ArmArch.Open
9/7 | PR [8024](https://runfo.azurewebsites.net/view/build/?number=8024) | net7.0-Linux-Debug-**arm64-Mono**_release-(Ubuntu.1804.ArmArch.Open)Ubuntu.1804.ArmArch.Open
9/7 | PR [6989](https://runfo.azurewebsites.net/view/build/?number=6989) | net7.0-Linux-Debug-**arm64-Mono**_release-(Debian.11.Arm64.Open)Ubuntu.1804.Armarch.Open
9/6 | PR [6677](https://runfo.azurewebsites.net/view/build/?number=6677) | net7.0-Linux-Debug-**arm64-Mono**_release-(Debian.11.Arm64.Open)Ubuntu.1804.Armarch.Open
8/26-9/6 |  | 25x Core Dumps on **arm64-Mono** - 1x Rolling run [1975580](https://runfo.azurewebsites.net/view/build/?number=1975580) (8/30)

Example of error ([Console log](https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-heads-main-7964053fabd4433ba3/System.Net.Security.Tests/1/console.10a48bc6.log?%3Fhelixlogtype%3Dresult)):
```
  Discovering: System.Net.Security.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  System.Net.Security.Tests (found 297 of 487 test cases)
  Starting:    System.Net.Security.Tests (parallel test collections = on, max threads = 6)
    System.Net.Security.Tests.SslStreamEKUTest.SslStream_SelfSignedClientEKUClientAuth_Ok [SKIP]
      Condition(s) not met: "IsRootCertificateInstalled"
    System.Net.Security.Tests.SslStreamEKUTest.SslStream_ServerEKUClientAuth_Fails [SKIP]
      Condition(s) not met: "IsRootCertificateInstalled"
    System.Net.Security.Tests.SslStreamEKUTest.SslStream_ClientEKUServerAuth_Fails [SKIP]
      Condition(s) not met: "IsRootCertificateInstalled"
    System.Net.Security.Tests.SslStreamEKUTest.SslStream_NoEKUServerAuth_Ok [SKIP]
      Condition(s) not met: "IsRootCertificateInstalled"
    System.Net.Security.Tests.SslStreamEKUTest.SslStream_NoEKUClientAuth_Ok [SKIP]
      Condition(s) not met: "IsRootCertificateInstalled"
    System.Net.Security.Tests.SslStreamCertificateTrustTest.SslStream_SendCertificateTrust_ThrowsOnUnsupportedPlatform [SKIP]
      Condition(s) not met: "DoesNotSupportSendingCustomCANamesInTls"
Process terminated.
Error while reaping child. errno = 10

=================================================================
	Native Crash Reporting
=================================================================
Got a SIGABRT while executing native code. This usually indicates
a fatal error in the mono runtime or one of the native libraries 
used by your application.
=================================================================

=================================================================
	Native stacktrace:
=================================================================
	0x7fb7424440 - Unknown

=================================================================
	External Debugger Dump:
=================================================================
[New LWP 25]
[New LWP 26]
```

#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|0|0|0|

### PR DESCRIPTION
**Pull request by wfurt**
duplicates #75424 for SslStream tests
fixes #75484

### CODE REVIEW

**Review by karelz — APPROVED** [review]

File: src/libraries/System.Net.Security/tests/FunctionalTests/TestHelper.cs [hunk 1]
```diff
@@ -260,5 +261,15 @@ static string ProtocolToString(SslProtocols? protocol)
 
             return name;
         }
+
+        [ModuleInitializer]
+        internal static void InitializeQuic()
```
Comment (jkotas) [inline src/libraries/System.Net.Security/tests/FunctionalTests/TestHelper.cs:266]: We should move this into a file under Common and just include that file into the tests that need this workaround.

**Review by danmoseley — APPROVED** [review]

File: src/libraries/System.Net.Security/tests/FunctionalTests/TestHelper.cs [hunk 2]
```diff
@@ -4,6 +4,7 @@
 using System.IO;
 using System.Linq;
 using System.Net.Sockets;
+using System.Net.Quic;
```
Comment (jkotas) [inline src/libraries/System.Net.Security/tests/FunctionalTests/TestHelper.cs:7]: Revert changes in this file

Comment (wfurt) [reply src/libraries/System.Net.Security/tests/FunctionalTests/TestHelper.cs]: done.

File: src/libraries/System.Net.Security/tests/FunctionalTests/TestHelper.cs [hunk 3]
```diff
@@ -260,5 +260,15 @@ static string ProtocolToString(SslProtocols? protocol)
...
+
+        [ModuleInitializer]
+        internal static void InitializeQuic()
+        {
+            // This will load Quic (if supported) to avoid interference with RemoteExecutor
+            // See https://github.com/dotnet/runtime/pull/75424 for more details
+            // IsSupported currently does not unload lttng. If it does in the future,
+            // we may need to call some real Quic API here to get everything loaded properly
+            _ = OperatingSystem.IsLinux() && QuicConnection.IsSupported;
+        }
```
Comment (jkotas) [inline src/libraries/System.Net.Security/tests/FunctionalTests/TestHelper.cs:272]:
```suggestion
```
This should be deleted too

File: src/libraries/System.Net.Mail/tests/Functional/System.Net.Mail.Functional.Tests.csproj [hunk 4]
```diff
@@ -24,6 +25,8 @@
              Link="Common\System\Diagnostics\Tracing\TestEventListener.cs" />
     <Compile Include="$(CommonTestPath)System\IO\TempFile.cs"
              Link="Common\System\IO\TempFile.cs" />
+    <Compile Include="$(CommonTestPath)System\Net\QuicLoad.cs"
```
Comment (jkotas) [inline src/libraries/System.Net.Mail/tests/Functional/System.Net.Mail.Functional.Tests.csproj:28]: Did we actually see "Error while reaping child" error in the System.Net.Mail tests or is this just a pre-emptive change?

Comment (wfurt) [reply src/libraries/System.Net.Mail/tests/Functional/System.Net.Mail.Functional.Tests.csproj:28]: no we did not. There are two failures I'm investigating so it is pre-emptive.

**Review by liveans — APPROVED** [review]

**Comment by karelz** [comment]
@CarnaViire @ManickaP can you please help with manual backport here? Thanks!

**Pull request merged**